### ISSUE TITLE
Wrong Error?

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#3136]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154, SC2154
- My shellcheck version (`shellcheck --version` or 'online'): 0.10.0
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [x] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

    do_on_exit END

    END()
    {
        kill "$tailpid" 2>/dev/null
        kill "$zenitypid" 2>/dev/null
    }


```

#### Here's what shellcheck currently says:
```
In bin/mailxmon line 14:
        kill "$tailpid" 2>/dev/null
        ^-------------------------^ SC2317 (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).


In bin/mailxmon line 15:
        kill "$zenitypid" 2>/dev/null
        ^---------------------------^ SC2317 (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

```
#### Here's what I wanted or expected to see:
```
Line 7	SC2154: tailpid is referenced but not assigned.
Line 8	SC2154: zenitypid is referenced but not assigned.
```

Looks like I am getting the wrong error using
```
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.10.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net

$ uname -a
Linux dev-10-34-32-2 4.18.0-553.32.1.el8_10.x86_64 #1 SMP Mon Dec 2 06:32:20 EST 2024 x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by wileyhy** [comment]
Is the function/alias 'do_on_exit' defined in the original script?

Wiley

**Comment by e-kwsm** [comment]
ShellCheck 0.11.0 issues SC2154.
